rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -568,6 +569,52 @@ def shadow_root(element)
           ShadowRoot.new self, shadow_root_id_from(id)
         end
 
+        #
+        # virtual-authenticator
+        #
+
+        def virtual_authenticator_id
+          @authenticator_id
+        end
+
+        def add_virtual_authenticator(options)
+          @authenticator_id = execute :add_virtual_authenticator, {}, options.to_dict
+        end
+
+        def remove_virtual_authenticator
+          execute :remove_virtual_authenticator, {}, {authenticatorId: @authenticator_id}
+          @authenticator_id = nil
+        end
+
+        def add_credential(credential)
+          credential = credential.to_dict
+          credential[:authenticatorId] = @authenticator_id
+          execute :add_credential, {}, credential
+        end
+
+        def credentials
+          credential_data = execute :get_credentials, {}, {authenticatorId: @authenticator_id}
+          credential_list = []
+
+          credential_data.each do |cred|
+            credential_list.push(Credential.new.from_dict(cred))
+          end
+          credential_list
```

Comment:
```suggestion
          credential_data.map do |cred|
            Credential.new.from_dict(cred)
          end
```

This would produce the same result.

Reply:
Changes incorporated. Thank you.